lib/LANraragi/Model/Tankoubon.pm

```diff
@@ -0,0 +1,270 @@
+package LANraragi::Model::Tankoubon;
+
+use feature qw(signatures);
+no warnings 'experimental::signatures';
+
+use experimental "try";
+
+use strict;
+use warnings;
+use utf8;
+
+use Redis;
+use Mojo::JSON qw(decode_json encode_json);
+use List::Util qw(min);
+
+use LANraragi::Utils::Database qw(redis_encode redis_decode invalidate_cache get_archive_json_multi);
+use LANraragi::Utils::Logging qw(get_logger);
+
+# get_tankoubon_list(page)
+#   Returns a list of all the Tankoubon objects.
+# NEEDS PAGINATION
+sub get_tankoubon_list($page=0) {
+
+    my $redis = LANraragi::Model::Config->get_redis;
+    my $logger = get_logger( "Tankoubon", "lanraragi" );
+
+    # Tankoubons are represented by RG_[timestamp] in DB. Can't wait for 2038!
+    my @tanks = $redis->keys('TANK_??????????');  
+
+    # Jam tanks into an array of hashes
+    my @result;
+    foreach my $key (sort @tanks) {
+        my %data = get_tankoubon($key);
+        push( @result, \%data );
+    }
+
+    # # Only get the first X keys
+    my $keysperpage = LANraragi::Model::Config->get_pagesize;
+
+    # Return total keys and the filtered ones
+    my $total = $#tanks+1;
+    my $start = $page*$keysperpage;
+    my $end = min( $start + $keysperpage - 1, $#result );
+    return ( $total, $#result + 1, @result[ $start .. $end ] );
+
+    #return @result;
+}
+
+# create_tankoubon(name, existing_id)
+#   Create a Tankoubon.
+#   If an existing Tankoubon ID is supplied, said Tankoubon will be updated with the given parameters.
+#   Returns the ID of the created/updated Tankoubon.
+sub create_tankoubon( $name, $tank_id ) {
+
+    my $redis = LANraragi::Model::Config->get_redis;
+
+    # Set all fields of the group object
+    unless ( length($tank_id) ) {
+        $tank_id = "TANK_" . time();
+
+        my $isnewkey = 0;
+        until ($isnewkey) {
+
+            # Check if the group ID exists, move timestamp further if it does
```

Comment:
According to the comment on top this sub should just update the name if given an existing tank ID, but instead it's generating a new ID?

Reply:
Not sure what you mean, but in this case I noted I didn't implement the edit with this(I was sending always tank_id as ""), so I added a few lines to edit the name.

Reply:
Ah yeah my mistake, I didn't read the `unless` statement properly, it'd only generate a new tank if tank_id was blank.  
It was still missing the edit/name update as mentioned though, yep!